vina_slam:
  ros__parameters:
    finish: false
    # ---------- General ----------
    General:
      lid_topic: "/livox/lidar"
      imu_topic: "/livox/imu"
      # The file path of the map to be saved R_opt_odom_LM  R_opt_odom_LM_LC
      save_path: "/home/yang/Code/vina_slam_ws/key_frame_map/"
      previous_map: "#HKisland03_R_opt: 0.3"
      # The bag name played in this run
      bagname: "zry_test"
      lidar_type: 0
      blind: 0.5
      point_filter_num: 1
      extrinsic_tran: [0.04165, 0.02326, -0.0284]
      extrinsic_rota: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]
      # The session is saved in save_path + bagname
      is_save_map: 0
      if_loop_detect: 0
      #0: with loop detect
      #1: no loop detect
    # ---------- Odometry ----------
    Odometry:
      cov_gyr: 0.01
      cov_acc: 1.0
      rdw_gyr: 0.0001
      rdw_acc: 0.0001
      down_size: 0.5
      dept_err: 0.01
      beam_err: 0.01
      voxel_size: 4.0
      min_eigen_value: 0.01
      degrade_bound: 100
      point_notime: 0

    # ---------- LocalBA ----------
    LocalBA:
      win_size: 10
      max_layer: 2
      cov_gyr: 0.01
      cov_acc: 1.0
      rdw_gyr: 0.0001
      rdw_acc: 0.0001
      min_ba_point: 1
      plane_eigen_value_thre: [4.0, 4.0, 4.0, 4.0]
      imu_coef: 0.0001
      thread_num: 5

    # ---------- Loop ----------
    Loop:
      jud_default: 0.4
      icp_eigval: 9.0
      ratio_drift: 0.01
      curr_halt: 10
      prev_halt: 10
      acsize: 2
      mgsize: 2
      isHighFly: 1

    # ---------- GBA ----------
    GBA:
      voxel_size: 15.0
      min_eigen_value: 10.0
      eigen_value_array: [2.0, 2.0, 2.0, 2.0]
      total_max_iter: 10
